mail_edit/models/mail_message.py

```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Therp BV <http://therp.nl>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openerp import api, models, SUPERUSER_ID, _
+from openerp.exceptions import except_orm
+
+
+class MailMessage(models.Model):
+    _inherit = 'mail.message'
+
+    @api.multi
+    def check_access_rule(self, operation):
+        if operation in ['write', 'unlink']:
+            try:
+                super(MailMessage, self).check_access_rule(operation)
+            except except_orm:
+                raise
+            else:
+                for rec in self:
+                    if rec.create_uid.id != self.env.uid and \
```

Comment:
I still disagree with this, now with an argument: An admin (someone from group_system, but not root) goes to settings/technical regularly to clean up some mail messages. Don't ask why that's not automated. Maybe it's only fishing out the bosses last late night mail, doesn't matter. But they won't be able to do that any more after installing this module, and there's no way around it like changing access rules or the like, which would be somehow acceptable.

I think that's highly unexpected, and software shouldn't surprise experienced users who have read the manual. So no, please do that on the client side.